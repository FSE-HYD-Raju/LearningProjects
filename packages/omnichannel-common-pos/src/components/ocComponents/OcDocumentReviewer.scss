.OcDocumentReviewer {
    &-wrapper {
        overflow: scroll;
        padding: 0 30px;
        &::before {
            background: linear-gradient(to bottom, white 0%, rgba(255, 255, 255, 0) 62%, rgba(255, 255, 255, 0) 100%);
            content: ' ';
            height: 50px;
            left: 0;
            position: absolute;
            top: 0;
            width: 100%;
        }
        &::after {
            background: linear-gradient(to top, white 0%, rgba(255, 255, 255, 0) 62%, rgba(255, 255, 255, 0) 100%);
            bottom: 0;
            content: ' ';
            height: 50px;
            left: 0;
            position: absolute;
            width: 100%;
        }
    }
}
