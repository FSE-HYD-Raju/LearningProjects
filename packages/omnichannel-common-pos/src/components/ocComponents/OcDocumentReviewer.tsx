import * as R from "react";
import cssns from "../../utils/cssnsConfig";
const React = cssns("OcDocumentReviewer").React as typeof R;

interface OcDocumentReviewerProps {
	document: { body?: string };
}

const OcDocumentReviewer: React.FC<OcDocumentReviewerProps> = props => {
	return (
		<div className="wrapper">
			{props.document &&
			props.document.body && (
				<div className="content">
					<div
						dangerouslySetInnerHTML={{
							__html: props.document.body
						}}
					/>
				</div>
			)}
		</div>
	);
};

export default OcDocumentReviewer;
export { OcDocumentReviewerProps };
