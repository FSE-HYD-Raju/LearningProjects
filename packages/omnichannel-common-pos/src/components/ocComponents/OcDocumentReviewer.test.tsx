import * as React from "react";
import toJson from "enzyme-to-json";
import { mount, shallow } from "enzyme";

import OcDocumentReviewer from "./OcDocumentReviewer";

describe("OcDocumentReviewer", () => {
	it("succeeds at shallow mount without props", () => {
		const props = {
			document: {}
		};
		const wrapper = shallow(<OcDocumentReviewer {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("succeeds at deep mount without props", () => {
		const props = {
			document: {}
		};
		const wrapper = mount(<OcDocumentReviewer {...props} />);
		expect(wrapper.find(".OcDocumentReviewer-wrapper").length).toEqual(1);
	});

	it("should be empty without props", () => {
		const props = {
			document: {}
		};
		const wrapper = mount(<OcDocumentReviewer {...props} />);
		expect(wrapper.find(".OcDocumentReviewer-wrapper").text()).toEqual("");
		expect(
			wrapper.find(".OcDocumentReviewer-wrapper").children().length
		).toEqual(0);
	});

	it("should include given simple html document", () => {
		const props = {
			document: {
				body: "<span>hi</span>"
			}
		};

		const wrapper = mount(<OcDocumentReviewer {...props} />);
		const dangerousHtml = wrapper
			.find(".OcDocumentReviewer-wrapper .OcDocumentReviewer-content")
			.childAt(0);
		expect(dangerousHtml.html()).toEqual("<div><span>hi</span></div>");
	});

	it("should include given html document with FormattedMessage element", () => {
		const props = {
			document: {
				body:
					"<FormattedMessage id='oc-label-1'description='Oc label child 1' defaultMessage='Child 1' />"
			}
		};
		const wrapper = mount(<OcDocumentReviewer {...props} />);
		const dangerousHtml = wrapper
			.find(".OcDocumentReviewer-wrapper .OcDocumentReviewer-content")
			.childAt(0);
		expect(dangerousHtml.html()).toEqual(
			"<div><formattedmessage id=\"oc-label-1\" description=\"Oc label child 1\" defaultmessage=\"Child 1\"></formattedmessage></div>"
		);
	});
});
