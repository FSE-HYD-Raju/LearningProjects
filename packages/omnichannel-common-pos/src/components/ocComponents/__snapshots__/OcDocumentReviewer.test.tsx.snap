// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcDocumentReviewer succeeds at shallow mount without props 1`] = `
<div
  className="OcDocumentReviewer-wrapper"
/>
`;
